index.d.ts

```diff
@@ -0,0 +1,419 @@
+    Menu: React.FunctionComponent<DropdownMenuProps>
+    Item: React.FunctionComponent<DropdownProps>
+  }
+
+  export interface FilterListProps extends CommonProps {
+    small?: boolean
+  }
+
+  export interface FilterListItemProps extends CommonProps {
+    count?: number
+    selected?: boolean
+  }
+
+  export const FilterList: React.FunctionComponent<FilterListProps> & {
+    Item: React.FunctionComponent<FilterListItemProps>
+  }
+
+  export interface FlashProps extends CommonProps {
+    full?: boolean
+    scheme?: string
+  }
+
+  export const Flash: React.FunctionComponent<FlashProps>
+
+  export interface CounterLabelProps extends CommonProps {
+    scheme?: string
+  }
+
+  export const CounterLabel: React.FunctionComponent<CounterLabelProps>
+
+  export interface LabelProps extends CommonProps {
+    outline?: boolean
+    size?: 'small' | 'medium' | 'large' | 'xl'
+    dropshadow?: boolean
+  }
+
+  export const Label: React.FunctionComponent<LabelProps>
+
+  export interface LinkProps extends CommonProps, TypographyProps {
+    href: string
+    underline?: boolean
+  }
+
+  export const Link: React.FunctionComponent<LinkProps>
+
+  export interface PointerBoxProps extends CommonProps, LayoutProps {
+    caret?: string
+  }
+
+  export const PointerBox: React.FunctionComponent<PointerBoxProps>
+
+  export interface PositionComponentProps extends PositionProps, CommonProps, LayoutProps {}
+
+  export const Relative: React.FunctionComponent<PositionComponentProps>
+  export const Absolute: React.FunctionComponent<PositionComponentProps>
+  export const Sticky: React.FunctionComponent<PositionComponentProps>
+  export const Fixed: React.FunctionComponent<PositionComponentProps>
+
+  export interface StateLabelProps extends CommonProps {
+    small?: boolean
+    status: 'issueOpened' | 'issueClosed' | 'pullOpened' | 'pullClosed' | 'pullMerged'
+  }
+
+  export const StateLabel: React.FunctionComponent<StateLabelProps>
+
+  export interface TextInputProps extends CommonProps {
+    autocomplete?: string
+    'aria-label'?: string
+    block?: boolean
+    disabled?: boolean
+    id?: string
+    name?: string
+    onChange?: (evt: React.ChangeEvent<HTMLInputElement>) => void
+    placeholder?: string
+    required?: boolean
+    size?: 'small' | 'large'
+    value?: string
+  }
+
+  export const TextInput: React.FunctionComponent<TextInputProps>
+
+  export interface TooltipProps extends CommonProps {
+    align?: 'left' | 'right'
+    direction?: 'n' | 'ne' | 'e' | 'se' | 's' | 'sw' | 'w' | 'nw'
+    noDelay?: boolean
+    text?: string
+    wrap?: boolean
+  }
+
+  export const Tooltip: React.FunctionComponent<TooltipProps>
+
+  export interface UnderlineNavProps extends CommonProps {
+    actions?: React.ReactNode
+    align?: 'right'
+    full?: boolean
+    label?: string
+  }
+
+  export const UnderlineNav: React.FunctionComponent<UnderlineNavProps>
+
+  export const theme: Object
```

Comment:
Note that the theme should be properly typed, but that would be a lot of effort 😅 I think this is good enough ™️ for a TypeScript MVP, and we can always improve this later.